### ISSUE TITLE
Error: "Configuration for rule "camelcase" is invalid"

### ISSUE THREAD

**Issue by r0mflip** [eslint/eslint#10994]
**Tell us about your environment**

* **ESLint Version:** v5.7.0
* **Node Version:** v10.11.0
* **npm Version:** 6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?**
`babel-eslint`

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "extends": "r0mflip",
  "env": {
    "node": true
  },
  "parserOptions": {
    "ecmaVersion": 2017
  },
  "rules": {
    "max-len": [2, {
      "code": 100,
      "tabWidth": 2,
      "ignoreUrls": true,
      "ignoreComments": true
    }]
  }
}
```

Using: https://github.com/r0mflip/eslint-config-r0mflip/blob/b08aa97821a398da9cd454378a1a6d14b55c3123/index.js#L198

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

``` json
// package.json
{
  "test": "eslint ."
}

```

```bash
npm test
```

**What did you expect to happen?**

ESLint run normally. I've taken code from https://eslint.org/docs/rules/camelcase#properties-always

**What actually happened? Please include the actual, raw output from ESLint.**
```
r0mflip:
	Configuration for rule "camelcase" is invalid:
	Value {"properties":"never","ignoreDestructuring":true} should NOT have additional properties.

Referenced from: /mnt/d/dev/git/r0mflip/serv/.eslintrc
Error: r0mflip:
	Configuration for rule "camelcase" is invalid:
	Value {"properties":"never","ignoreDestructuring":true} should NOT have additional properties.
```

**Comment by nzakas** [comment]
Could not reproduce using ESLint v.5.7.0. Is it possible that you're using an older version of ESLint? The "ignoreDestructuring" property was added in v5.0.0, and you would definitely get this warning if you were using an older version of ESLint.

**Comment by morganney** [comment]
I'm experiencing a similar error with this configuration:
```
camelcase: [
  'error',
   {
      allow: ['id_token']
   }
]
```
The error message is:
```
	Configuration for rule "camelcase" is invalid:
	Value {"allow":["id_token"]} should NOT have additional properties.
```

ESLint version: v5.14.1
Node version: v10.15.1
babel-eslint version: 8.2.6 (also tried 10.0.1)

**Comment by platinumazure** [comment]
@morganney Could you please open a new issue? This looks different than the originally reported issue. Thanks!
